test/server.jl

```diff
@@ -168,4 +168,18 @@ end
     @test_skip HTTP.hasheader(HTTP.get("http://httpbin.org/redirect-to?url=https://httpbin.org/response-headers?Authorization=auth"), "Authorization")
 end # @testset
 
+@testset "on_shutdown" begin
+    # shutdown unit tests
+    @test HTTP.Servers.shutdown(nothing) === nothing
+    a = 0
+    shutdown_add() = a += 1
+    HTTP.Servers.shutdown(shutdown_add)
+    @test a == 1
+    HTTP.Servers.shutdown([shutdown_add, shutdown_add])
+    @test a == 3
+    shutdown_throw() = throw(ErrorException("Broken"))
+    @test_logs (:error, r"hutdown function shutdown_throw failed") HTTP.Servers.shutdown([shutdown_throw, shutdown_add])
```

Comment:
```suggestion
    @test_logs (:error, r"shutdown function shutdown_throw failed") HTTP.Servers.shutdown([shutdown_throw, shutdown_add])
```

Reply:
Also looks like this test fails on 1.0

Reply:
The exact printing of function can change so perhaps just
```suggestion
    @test_logs (:error, r"hutdown function .* failed") HTTP.Servers.shutdown([shutdown_throw, shutdown_add])
```

Reply:
Yep that's nice, thanks